folding-schemes/src/transcript/mod.rs

```diff
@@ -1,31 +1,62 @@
-use crate::Error;
+use ark_crypto_primitives::sponge::{constraints::CryptographicSpongeVar, CryptographicSponge};
 use ark_ec::CurveGroup;
 use ark_ff::PrimeField;
-use ark_r1cs_std::{boolean::Boolean, fields::fp::FpVar};
-use ark_relations::r1cs::{ConstraintSystemRef, SynthesisError};
-use ark_std::fmt::Debug;
+use ark_r1cs_std::{
+    boolean::Boolean, fields::fp::FpVar, groups::CurveVar, ToConstraintFieldGadget,
+};
+use ark_relations::r1cs::SynthesisError;
 
 pub mod poseidon;
 
-pub trait Transcript<C: CurveGroup> {
-    type TranscriptConfig: Debug;
+pub trait AbsorbNonNative<F: PrimeField> {
+    /// Converts the object into field elements that can be absorbed by a `CryptographicSponge`.
+    /// Append the list to `dest`
+    fn to_native_sponge_field_elements(&self, dest: &mut Vec<F>);
```

Comment:
One question, I've tried to find where the `to_native_sponge_field_elements` is used and haven't succeed (other than structs implementing the method to implement the `AbsorbNonNative` trait). The usages of the `AbsorbNonNative` trait that I've found use the one that returns the `Vec<F>` instead of modifying a given `dest` input value.
Does the `AbsorbNonNative` have the method with the `dest` param in order to match the native `Absorb` trait, which internally uses a similar method from arkworks (also modifying the `dest` param)?